examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPoolPublisher.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomPayloadPoolPublisher.h
+ *
+ */
+
+#ifndef CUSTOM_PAYLOAD_POOL_PUBLISHER_H_
+#define CUSTOM_PAYLOAD_POOL_PUBLISHER_H_
+
+#include "CustomPayloadPoolPubSubTypes.h"
+#include "PayloadPool.hpp"
+
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+
+class CustomPayloadPoolPublisher : public eprosima::fastdds::dds::DataWriterListener
+{
+public:
+
+    CustomPayloadPoolPublisher();
+
+    ~CustomPayloadPoolPublisher();
+
+    //!Initialize
+    bool init();
+
+    //!Publish a sample
+    bool publish(
+            bool waitForListener = true);
```

Comment:
Snake case all over please. Applies to other variables as well
```suggestion
            bool wait_for_listener = true);
```